### PR TITLE
Feature/saltstack master node creating v1

### PR DESCRIPTION
**Pull request by freideveloper**
added a new docker file for salt-master

### CODE REVIEW

**Comment by freideveloper** [comment]
please review @Andreichenko

File: salt-master/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+RUN yum install -y openssh-server \
+    openssh-clients
+
+RUN yum clean all
+
+# SSHD settings for root login and install
+RUN sed -i -e "s|^#PermitRootLogin .*$|PermitRootLogin yes|" /etc/ssh/sshd_config \
+    && sed 's@session\*required\*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+
```
Comment (Andreichenko) [inline salt-master/Dockerfile:51]: OK

File: salt-master/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
 COPY setup.sh /opt/setup.sh
+EXPOSE 4505 4506
```
Comment (Andreichenko) [inline salt-master/Dockerfile]: need to remove

Comment (Andreichenko) [reply salt-master/Dockerfile]: what about cgroup?

File: salt-master/Dockerfile [hunk 3]
```diff
@@ -6,17 +6,19 @@ MAINTAINER Sascha F. Andreichenko <andreichenko@web.de>
 RUN curl -L https://bootstrap.saltstack.com -o install_salt.sh
 ###### RUN script for salt
 RUN sh install_salt.sh -P -M -X
-RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done);
-rm -f /lib/systemd/system/multi-user.target.wants/*;\
-rm -f /etc/systemd/system/*.wants/*;\
```
Comment (Andreichenko) [inline salt-master/Dockerfile]: OK

File: salt-master/Dockerfile [hunk 4]
```diff
@@ -6,17 +6,19 @@ MAINTAINER Sascha F. Andreichenko <andreichenko@web.de>
...
+RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done); \
+rm -f /lib/systemd/system/multi-user.target.wants/*; \
+rm -f /etc/systemd/system/*.wants/*; \
```
Comment (Andreichenko) [inline salt-master/Dockerfile:11]: OK

File: salt-master/Dockerfile [hunk 5]
```diff
@@ -6,17 +6,19 @@ MAINTAINER Sascha F. Andreichenko <andreichenko@web.de>
...
 rm -f /lib/systemd/system/local-fs.target.wants/*; \
 rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
 rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
 rm -f /lib/systemd/system/basic.target.wants/*;\
 rm -f /lib/systemd/system/anaconda.target.wants/*;
 
+#mount cgroup
```
Comment (Andreichenko) [inline salt-master/Dockerfile:18]: OK

File: salt-master/Dockerfile [hunk 6]
```diff
@@ -26,7 +28,24 @@ RUN yum clean all
...
+
+# salt files
+RUN mkdir -p /srv/salt
+RUN mkdir -p /srv/pillar
+
+# change the root's password
+RUN echo "root:password" | chpasswd
+
+# ready for using
+CMD ["/usr/sbin/init"]
```
Comment (Andreichenko) [inline salt-master/Dockerfile:50]: OK

**Review by Andreichenko — APPROVED** [review]
docker file has been tested on preprod, we need to create a staging env

**Pull request merged**